src/main/java/ch/njol/skript/sections/SecLoop.java

```diff
@@ -129,12 +132,21 @@ public SecLoop setNext(@Nullable TriggerItem next) {
 	}
 
 	@Nullable
+	@Override
 	public TriggerItem getActualNext() {
 		return actualNext;
 	}
 
-	public void exit(Event event) {
-		current.remove(event);
-		currentIter.remove(event);
+	@Override
+	public void exit(Event e) {
+		current.remove(e);
+		currentIter.remove(e);
+		currentLoopCounter.remove(e);
+	}
+
+	@Override
+	public long getLoopCounter(Event e) {
+		return currentLoopCounter.getOrDefault(e, 1L);
```

Comment:
loop counter must be at least of value 1, in the getLoopCounter methods I used `getOrDefault(e, 1L)` but in the `walk` method I used `getOrDefault(e, 0L)` and then added 1 to it because we're increasing the value 1 per walk and getting a default value of 1 instead of 0 in walk method would make the first iteration be at value of 2